### PR TITLE
Add Roslyn

### PR DESCRIPTION
**Pull request by ironcev**
https://github.com/ironcev/awesome-roslyn

A curated list of Roslyn books, tutorials, open source projects, analyzers, code fixes, and refactorings.

Roslyn is C#/VB.NET Compiler-as-a-Service and as such it is awesome in its own :-) This list captures awesome learning materials as well as awesome open source projects built on top of Roslyn.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Hey, sorry for the lack of response, please see #1363.

File: readme.md [hunk 1]
```diff
@@ -152,6 +152,7 @@
 - [Fortran](https://github.com/rabbiabram/awesome-fortran)
 - [.NET](https://github.com/quozd/awesome-dotnet)
 	- [Core](https://github.com/thangchung/awesome-dotnet-core)
+	- [Roslyn](https://github.com/ironcev/awesome-roslyn)
```
Comment (aschrijver) [inline readme.md]: There was rule that the URL should end with `#readme`. Not sure if it still applies.

**Comment by aschrijver** [comment]
Hi @ironcev 

I reviewed your PR. Your list overall looks quite good, but I have 1 suggestion on https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md and a remark

- Did you review 2 other PR's to this repo? That is a requirement for inclusion to lighten the work for @sindresorhus You can include a link to the reviewed PR's in a comment
- Not sure if this is still a requirement, but for a short time there was the rule that the awesome URL entry should have a `#readme` hash at the end

**Comment by ironcev** [comment]
Hi @aschrijver ,

Thanks a lot for your review. I am glad to hear that the list overall looks quite good :-) I put quite some effort into it. Regarding the mentioned points, for the first one, yes I am aware of the fact that I should do two reviews and I will gladly and willingly do them. I fully support @sindresorhus initiative to share the load of reviewing and I hope the community will start doing it. But before doing the reviews I want to fix one or two smaller findings that I found on my own after submitting the PR like e.g. the image does not link to the project site.

Regarding the second point, I am not aware of that requirement, or at least I didn't see it mentioned anywhere.

Thanks again for your review! My next step will be to do the mentioned fixes, do two reviews and also later to contribute as a reviewer. I admire the effort behind the Awesome list and want to contribute more.

**Comment by sindresorhus** [comment]
```
❯ awesome-lint https://github.com/ironcev/awesome-roslyn
✖ Linting

  readme.md:1:1
  ✖   1:1    Missing newline character at end of file                      remark-lint:final-newline
  ✖  23:64   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  24:90   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  25:86   List item link and description must be separated with a dash  remark-lint:awesome/list-item
  ✖  38:51   List item description must start with valid casing            remark-lint:awesome/list-item
  ✖  39:66   List item description must start with valid casing            remark-lint:awesome/list-item
  ✖  54:74   List item description must start with valid casing            remark-lint:awesome/list-item
  ✖  64:102  List item description must start with valid casing            remark-lint:awesome/list-item

  8 errors
```

https://github.com/sindresorhus/awesome-lint

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by sindresorhus** [comment]
> yes I am aware of the fact that I should do two reviews and I will gladly and willingly do them.

:arrow_up:

**Comment by sindresorhus** [comment]
> Includes a project logo/illustration whenever possible.
**The image should link to the project website or any relevant website.
The image should be high-DPI. Set it to maximum half the width of the original image.**

**Comment by ironcev** [comment]
Thanks for the detailed instructions @sindresorhus! 

At the time I've created this checklist, compiled from [Creating Your Own List](https://github.com/sindresorhus/awesome/blob/master/create-list.md), [The awesome manifesto](https://github.com/sindresorhus/awesome/blob/master/awesome.md), and the [Awesome pull request template](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md). (The Awesome Linter point is added now.) I've used it for checking my list and planned to use it for reviewing other lists. My list still doesn't fully passes all the below checks e.g. the image not linking to the relevant website.

I'll take time over the weekend to make the list fully compliant, fix the merge issue, and review two other lists.

- [ ] Search this list before making a new one, as yours may be a duplicate.
- [ ] Your awesome list should be named awesome-list-name and have the awesome badge, a license, and contribution guidelines.
- [ ] Wait at least 30 days after creating a list before submitting it.
- [ ] Research if the stuff you're including is actually awesome. Only put stuff on the list that you or another contributor can personally recommend.
- [ ] Inform your readers why it's on the list and how they will benefit from it.
- [ ] Have a succinct description at the top of your readme. Make sure your list covers a certain scope and nothing else.
- [ ] Link to other awesome lists if you think they already cover a certain subject well enough.
- [ ] List is grammatically correct, typo-free and has no Markdown formatting errors.
- [ ] Include contribution guidelines.
- [ ] Create a table of contents, organize the content into different categories, and use images if suitable. 
- [ ] Ensure all entries are consistent (e.g. all entry descriptions end in a .).
- [ ] Review at least 2 other open pull requests.
- [ ] The pull request has a descriptive title in the format Add Name of List (Example: Add Swift), not Update readme.md or Add awesome list.
- [ ] The entry in the Awesome list should:
  - [ ] Include a short description about the project/theme of the list. It should not describe the list itself. Example: - [Fish](…) - User-friendly shell., not - [Fish](…) - Resources for Fish..
  - [ ] Be added at the bottom of the appropriate category.
- [ ] Non-generated Markdown file in a GitHub repo.
- [ ] Includes a succinct description of the project/theme at the top of the readme.
- [ ] The repo should have awesome-list & awesome as GitHub topics. I encourage you to add more relevant topics.
- [ ] Not a duplicate.
- [ ] Includes a project logo/illustration whenever possible.
  - [ ] Either centered, fullwidth, or placed at the top-right of the readme.
  - [ ] The image should link to the project website or any relevant website.
  - [ ] The image should be high-DPI. Set it to maximum half the width of the original image.
- [ ] Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- [ ] Includes the Awesome badge.
  - [ ] Should be placed on the right side of the readme heading.
  - [ ] Can be placed centered if the list has a centered graphics header.
  - [ ] Should link back to this list.
- [ ] Has a Table of Contents section.
  - [ ] Should be named Contents, not Table of Contents.
  - [ ] Should be the first section in the list.
  - [ ] Should only have one level of nested lists, preferably none.
- [ ] Has an appropriate license.
  - [ ] That means something like CC0, not a code licence like MIT, BSD, Apache, etc.
  - [ ] WTFPL and Unlicense are not acceptable licenses.
  - [ ] If you use a license badge, it should be SVG, not PNG.
- [ ] Has contribution guidelines.
  - [ ] The file should be named contributing.md. Casing is up to you.
- [ ] Has consistent formatting and proper spelling/grammar.
  - [ ] The link and description are separated by a dash. Example: - [AVA](…) - JavaScript test runner.
  - [ ] The description starts with an uppercase character and ends with a period.
  - [ ] Consistent and correct naming. For example, Node.js, not NodeJS or node.js.
- [ ] Doesn't include a Travis badge.
- [ ] Passes all the [Awesome Lint](https://github.com/sindresorhus/awesome-lint) checks.

**Comment by ironcev** [comment]
Hi @sindresorhus,

All of the criteria listed in the above comment are now fulfilled. The linter still shows four "List item description must start with valid casing" errors, but this is [a known bug in the linter](https://github.com/sindresorhus/awesome-lint/issues/52#issuecomment-468796294) (_C#_ is not recognized as a valid casing for the beginning of a list item description).

In addition, I've reviewed the following pull requests:
#1091 
#1533 
#1513

**Comment by sindresorhus** [comment]
This is not done: https://github.com/sindresorhus/awesome/pull/1317#issuecomment-467332579

**Comment by sindresorhus** [comment]
The diff is weird: https://github.com/sindresorhus/awesome/pull/1317/files

File: readme.md (hunk 1 above)
Comment (sindresorhus) [reply readme.md]: Yes, should end in `#readme`

Comment (sindresorhus) [reply readme.md]: > The entry in the Awesome list should:
> Include a short description about the project/theme of the list. **It should not describe the list itself.**

**Comment by sindresorhus** [comment]
Ping

**Comment by ironcev** [comment]
Sorry for a lack of response and thanks for the reminder! Buried with a bunch of tasks, I completely forgot about this thread :-(  I'll check the remaining comments and provide input/fixes latest by the end of this week. Thanks and sorry once again!

**Comment by ironcev** [comment]
@aschrijver, @sindresorhus thanks for your reviews. Here is my feedback on all the topics that were open.

> There was rule that the URL should end with #readme. Not sure if it still applies.

The rule wasn't there when I was doing the original commit. Also, only very few entries in the list really ended with #readme. I see this is fixed now anywhere. Great since the rule definitely makes sense :-) **Fixed in my PR as well.** ✔️ 

> The entry in the Awesome list should: Include a short description about the project/theme of the list.

Again, not all entries have it and in particular, my group (.NET and Core) do not have it so I didn't want to stick out. True, *Roslyn* is a household name in the .NET world but not outside of it, so **I added the description**. ✔️ 

> This is not done: #1317 (comment)

Means "The image should link to the project website or any relevant website. The image should be high-DPI. Set it to maximum half the width of the original image."
Both things were fixed back in March with https://github.com/ironcev/awesome-roslyn/commit/5212901bf3002dae9de4c14e3c0641e74fc04e4a and https://github.com/ironcev/awesome-roslyn/commit/688f054845f5a346bc8c99278fc4f9687bc49d03. The image links to the Roslyn Github page and the resolution is 400x400 DPI which I consider to be high-DPI. The largest possible would be 500x500 DPI since the image is 1000px wide.

**So I don't understand what should still be done here.** ❓ 

> The diff is weird: https://github.com/sindresorhus/awesome/pull/1317/files

Yes, it was **very** weird! I have no clue what I did, anyhow **it's fixed now**. ✔️

**Comment by ironcev** [comment]
Thanks for the clarification @sindresorhus, width fixed to 500px. ✔️

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1127796609310322688

**Comment by sindresorhus** [comment]
I would recommend removing the text heading now `Awesome Roslyn`, since it looks weird having both it and the logo with the same text.

**Comment by ironcev** [comment]
It makes perfectly sense, thanks for the suggestion. I've removed the heading and it looks better now :-)

**Pull request merged**